purchases/src/main/kotlin/com/revenuecat/purchases/CustomerInfoHelper.kt

```diff
@@ -121,14 +125,30 @@ internal class CustomerInfoHelper(
             appInBackground,
             { info ->
                 log(LogIntent.RC_SUCCESS, CustomerInfoStrings.CUSTOMERINFO_UPDATED_FROM_NETWORK)
+                offlineEntitlementsManager.resetOfflineCustomerInfoCache()
                 cacheCustomerInfo(info)
                 sendUpdatedCustomerInfoToDelegateIfChanged(info)
                 dispatch { callback?.onReceived(info) }
             },
-            { error, _ ->
+            { error, isServerError ->
                 Log.e("Purchases", "Error fetching customer data: $error")
                 deviceCache.clearCustomerInfoCacheTimestamp(appUserID)
-                dispatch { callback?.onError(error) }
+                if (shouldCalculateOfflineCustomerInfo(isServerError, appUserID)) {
```

Comment:
`shouldCalculateOfflineCustomerInfo` is calling the `shouldCalculateOfflineCustomerInfoInGetCustomerInfoRequest`. I think the private function is hiding the `...GetCustomerInfoRequest` part of the call. Not sure if you had any intention with it, but I think it could be more clear to not have the private function, but your call